### PR TITLE
Fixed error on empty fields on /saveClinicalDetails

### PR DESCRIPTION
**Pull request by y-golde**
# the issue
With the addition of #1115 the state of an empty isolationAddress  was { city : '', street : '' , floor : '' , housenum : ''}
the problem was that in the Address table the fields city and street were nullable and had FK constraint to their respective table.
# the solution
added a new formatting function `formatToNullable` that takes those fields and formats empty fields to nulls
reasoning is that now the form is a generic components and this is the only exception where those fields need to be either nullable or FK.

fixes [1372](https://dev.azure.com/spectrumFactory/CoronaI/_workitems/edit/1372/)

### CODE REVIEW

File: server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts [hunk 1]
```diff
@@ -227,10 +226,10 @@ clinicalDetailsRoute.post('/saveClinicalDetails', (request: Request, response: R
         });
     } else {
         const updateAddressLogger = logger.setup({...logData, workflow: `${logData.workflow}: update isolation address`})
-
+        
         const parameters = {id: isolationAddressId, addressPatch: isolationAddress};
         updateAddressLogger.info(launchingDBRequestLog(parameters), Severity.LOW);
-
+        console.log(parameters);
```
Comment (roymaniv) [inline server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts:232]: remove console log

Comment (y-golde) [reply server/src/ClientToDBAPI/ClinicalDetailsRoute/mainRoute.ts]: removed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1130/commits/47d313eeb200710f20ceb2720e3a1810d9f7d97a)

**Review by roymaniv — APPROVED** [review]

File: server/src/Utils/addressUtils.ts [hunk 2]
```diff
@@ -6,4 +6,13 @@ export const formatToInsertAndGetAddressIdInput = (address: Address) :  InsertAn
     streetValue: address.street || null,
     floorValue: address.floor || null,
     houseNumValue: address.houseNum || null,
-});
\ No newline at end of file
+});
+
+export const formatToNullable = (address : Address): Address => {
+    return {
+        city : address.city !== '' ? address.city : null,
```
Comment (MayaSasson) [inline server/src/Utils/addressUtils.ts:13]: use the  || syntax

Comment (y-golde) [reply server/src/Utils/addressUtils.ts:13]: but I want to keep the original address in that its not null

Comment (y-golde) [reply server/src/Utils/addressUtils.ts:13]: I mean the original address is not null/undefined , it's ''

Comment (y-golde) [reply server/src/Utils/addressUtils.ts]: **UPDATE** : [it works](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/1130/commits/dbd2a677252a58d566a126c0b9a8c879a633f5a1) who knew, thanks Javascript!

**Review by ItayBenMoshe — APPROVED** [review]

**Review by MayaSasson — APPROVED** [review]

**Pull request merged**